planner/optimizer/src/main/java/eu/seaclouds/platform/planner/optimizer/CloudOffer.java

```diff
@@ -91,7 +97,8 @@ public void setCost(double cost) {
 
    @Override
    public CloudOffer clone() {
```

Comment:
I completely agree with you. There is the opinion that Java clone() is broken and should not be used. 
I took the issue of the objects references into account. In fact, the clone() method in class SuitableOptions (in this same package) cares about doing the deep clone. Actually, this clone() is only called by the clone in SuitableOptions class, while doing the deep copy. If you follow the opinion that clone() method is broken and we must avoid its utilization, I'll create a constructor that receives as parameter an object of the same class. 

Let me know if your comment was an opinion to discuss or a command to me to change it ;) .